### PR TITLE
Expose caret position and other APIs from EmojiTextArea

### PR DESCRIPTION
**Pull request by abhinayagarwal**

### CODE REVIEW

**Review by jperedadnr — COMMENTED** [review]
Looks good, just one comment

File: EmojiTextArea/src/main/java/com/gluonhq/emoji/control/EmojiTextArea.java [hunk 1]
```diff
@@ -3,79 +3,124 @@
...
+
+
     public void clear() {
         if (!text.isBound()) {
             setText("");
         }
     }
 
+    public void insertText(int position, String text) {
+        textarea.insertText(position, text);
```
Comment (jperedadnr) [inline EmojiTextArea/src/main/java/com/gluonhq/emoji/control/EmojiTextArea.java:109]: You should check if `textarea` is null (skin could not be created yet or user overrides `createDefaultSkin`

File: EmojiTextArea/pom.xml [hunk 2]
```diff
@@ -23,14 +23,6 @@
                 <configuration>
                     <release>11</release>
                     <compilerArgs>
-                        <arg>--add-opens</arg>
```
Comment (jperedadnr) [inline EmojiTextArea/pom.xml:26]: did you change RichTextFX dependency/version?

Comment (abhinayagarwal) [reply EmojiTextArea/pom.xml:26]: Yes, please check [here](https://github.com/gluonhq/wave-app/pull/200/files#diff-d91e51b39d00c9f8d4b3ac0f34c5fb717e99452165a8fff208131ee6b13beec2R92)

**Review by jperedadnr — APPROVED** [review]

**Pull request merged**